src/plots/cartesian/layout_attributes.js

```diff
@@ -465,6 +465,36 @@ module.exports = {
             'Only has an effect if `anchor` is set to *free*.'
         ].join(' ')
     },
+    categorymode: {
+        valType: 'enumerated',
+        values: [
+            'trace', 'category ascending', 'category descending', 'array'
```

Comment:
@mdtusz I agree with your point. The reason it went this way was because there was discussion on the issue about eventually adding value-based sort options, e.g. `value ascending` and probably it's good to keep the API stable. I initially used it with a dash but saw that the plotly pattern is multiword strings. Again, I agree with your sentiment and glad to replace it with a single word, just sharing the motives.